homeassistant/components/wallbox/select.py

```diff
@@ -0,0 +1,86 @@
+"""Home Assistant component for accessing the Wallbox Portal API. The switch component creates a switch entity."""
+
+from __future__ import annotations
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import (
+    CHARGER_DATA_KEY,
+    CHARGER_ECO_SMART_KEY,
+    CHARGER_FEATURES_KEY,
+    CHARGER_PLAN_KEY,
+    CHARGER_POWER_BOOST_KEY,
+    CHARGER_SERIAL_NUMBER_KEY,
+    CHARGER_SOLAR_CHARGING_MODE,
+    DOMAIN,
+    EcoSmartMode,
+)
+from .coordinator import WallboxCoordinator
+from .entity import WallboxEntity
+
+SELECT_TYPES: dict[str, SelectEntityDescription] = {
+    CHARGER_ECO_SMART_KEY: SelectEntityDescription(
+        key=CHARGER_ECO_SMART_KEY,
+        translation_key="eco_smart",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Create wallbox select entities in HASS."""
+    coordinator: WallboxCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [WallboxSelect(coordinator, SELECT_TYPES[CHARGER_ECO_SMART_KEY])]
+    )
```

Comment:
Since `SELECT_TYPES` is a dict, I kinda expected the code here to iterate over it (instead of hardcode picking an item from it.

```suggestion
    async_add_entities(
        [WallboxSelect(coordinator, description)]
        for description in SELECT_TYPES
    )
```

Reply:
Fixed this: https://github.com/home-assistant/core/pull/139286/commits/6fb968185dc7739cfaa8e3f94db1424993b2db0c

Your suggested code gave me an error because it only returns the key and not the value. I used the same coding style as other entities in the Wallbox component (see [number.py](https://github.com/home-assistant/core/blob/dev/homeassistant/components/wallbox/number.py#L102-L104) and [lock.py](https://github.com/home-assistant/core/blob/dev/homeassistant/components/wallbox/lock.py#L48-L50).

Note: in [switch.py](https://github.com/home-assistant/core/blob/dev/homeassistant/components/wallbox/switch.py#L39), which I used as base for the select.py file, it's also picking a value hardcoded. I'll fix that in a different PR.